src/caching/github-actions-cache.ts

```diff
@@ -652,6 +652,11 @@ function readBody(res: http.IncomingMessage): Promise<string> {
     res.on('error', (error: Error) => {
       reject(error);
     });
+    res.socket.on('close', (hadError) => {
+      if (hadError) {
+        reject(new Error('socket closed with error'));
```

Comment:
I'm not sure this will be sufficient, since we don't currently try/catch the call to `readBody`. So I think we also need to add checks for this failure everywhere `readBody` is called, and then do the `_maybeHandleServiceDown` thing on failure. Converting this API to use `Result` instead of exceptions would also be cool.

Reply:
Agree, but my goal is to first get the ECONNRESET errors to be in our stack trace rather than in the uncaught exceptions handler